base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -780,6 +801,142 @@ where
         Ok(())
     }
 
+    /// broadcasts a SHA-XTR atomic swap transaction
+    /// # Arguments
+    /// 'dest_pubkey': The Comms pubkey of the recipient node
+    /// 'amount': The amount of Tari to send to the recipient
+    /// 'fee_per_gram': The amount of fee per transaction gram to be included in transaction
+    pub async fn broadcast_sha_atomic_swap_transaction(
```

Comment:
Why does this need a specific broadcast?

Reply:
I can probably think of a better name here, but it does the creation of the HTLC and then the broadcast.

Reply:
renamed this to `send_sha_atomic_swap_transaction`